test/unix/force_merge.ml

```diff
@@ -0,0 +1,218 @@
+let reporter ?(prefix = "") () =
+  let report src level ~over k msgf =
+    let k _ =
+      over ();
+      k ()
+    in
+    let ppf = match level with Logs.App -> Fmt.stdout | _ -> Fmt.stderr in
+    let with_stamp h _tags k fmt =
+      let dt = Unix.gettimeofday () in
+      Fmt.kpf k ppf
+        ("%s%+04.0fus %a %a @[" ^^ fmt ^^ "@]@.")
+        prefix dt
+        Fmt.(styled `Magenta string)
+        (Logs.Src.name src) Logs_fmt.pp_header (level, h)
+    in
+    msgf @@ fun ?header ?tags fmt -> with_stamp header tags k fmt
+  in
+  { Logs.report }
+
+let string_size = 20
+
+let index_size = 100
+
+let () = Random.self_init ()
+
+let random_char () = char_of_int (33 + Random.int 94)
+
+let random_string () = String.init string_size (fun _i -> random_char ())
+
+module Key = struct
+  type t = string
+
+  let v = random_string
+
+  let hash = Hashtbl.hash
+
+  let hash_size = 10
+
+  let encode s = s
+
+  let decode s off = String.sub s off string_size
+
+  let encoded_size = string_size
+
+  let equal = String.equal
+
+  let pp s = Fmt.fmt "%s" s
+end
+
+module Value = struct
+  type t = string
+
+  let v = random_string
+
+  let encode s = s
+
+  let decode s off = String.sub s off string_size
+
+  let encoded_size = string_size
+
+  let pp s = Fmt.fmt "%s" s
+end
+
+module Index = Index_unix.Make (Key) (Value)
+
+let index_name = "another"
+
+let log_size = 4
+
+let t = Index.v ~fresh:true ~log_size index_name
+
+(* [tbl] is the in-memory representation of the [t], i.e. tbl should always
+contain the same elements as [t] *)
+let tbl =
+  let tbl = Hashtbl.create 0 in
+  let rec loop i =
+    if i = 0 then (
+      Index.flush t;
+      tbl )
+    else
+      let k = Key.v () in
+      let v = Value.v () in
+      Index.replace t k v;
+      Hashtbl.add tbl k v;
+      loop (i - 1)
+  in
+  loop index_size
```

Comment:
To avoid code duplication, the prelude of that file should probably be refactored in a different `common.ml` or `prelude.ml` file and be shared between all the tests.